Tasks/AzureFunctionOnKubernetesV0/task.json

```diff
@@ -0,0 +1,123 @@
+    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/az-functions-k8s",
+    "helpMarkDown": "[Learn more about this task](https://go.microsoft.com/fwlink/?linkid=848006) or [see Azure Functions on Kubernetes with KEDA](https://docs.microsoft.com/en-us/azure/azure-functions/functions-kubernetes-keda) or [check out sample-hello-world-azure-functions](https://github.com/kedacore/sample-hello-world-azure-functions)",
+    "category": "Deploy",
+    "preview": true,
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 0,
+        "Minor": 159,
+        "Patch": 0
+    },
+    "demands": [],
+    "groups": [
+        {
+            "name": "serviceConnections",
+            "displayName": "Service Connections",
+            "isExpanded": true
+        },
+        {
+            "name": "commands",
+            "displayName": "Commands",
+            "isExpanded": true
+        }
+    ],
+    "inputs": [
+        {
+            "name": "dockerRegistryServiceConnection",
+            "type": "connectedService:dockerregistry",
+            "label": "Docker registry service connection",
+            "groupName": "serviceConnections",
+            "helpMarkDown": "Select a Docker registry service connection."
+        },
+        {
+            "name": "kubernetesServiceConnection",
+            "type": "connectedService:kubernetes",
+            "label": "Kubernetes service connection",
+            "required": true,
+            "groupName": "serviceConnections",
+            "helpMarkDown": "Select a Kubernetes service connection."
+        },
+        {
+            "name": "secretName",
+            "type": "string",
+            "label": "Secret Name",
+            "groupName": "commands",
+            "helpMarkDown": "Kubernetes secret containing function config data (for ex. AzureWebJobsStorage: <Azure storage connection string>)"
+        },
+        {
+            "name": "namespace",
+            "type": "string",
+            "label": "Kubernetes namespace",
+            "groupName": "commands",
+            "helpMarkDown": "Kubernetes namespace"
+        },
+        {
+            "name": "appName",
+            "type": "string",
+            "required": true,
+            "label": "Application Name",
+            "groupName": "commands",
+            "helpMarkDown": "Application Name. The Kubernetes objects created use this name. This should follow Kubernetes naming conventions for resource names."
+        },
+        {
+            "name": "imageName",
+            "type": "string",
+            "label": "Image Name",
+            "groupName": "commands",
+            "helpMarkDown": "Image Name. If provided, the image is used for deployment. Otherwise, build and push (from the Dockerfile in the function root directory) and use that image."
+        },
+        {
+            "name": "registry",
```

Comment:
Change this as taking registry name for ACR is redundant as the same can be derived from docker registry service connection as well. Make it consistent with Docker@2 and have the following inputs -
- dockerRegistryServiceConnection
- image (rename imageName) to image

So if user wants to push to his own private repository on DockerHub, he'll use the SC for specifying that the registry to use is Docker Hub and the image input is of the form <account-name>/<repository>

Reply:
![image](https://user-images.githubusercontent.com/11447401/65572292-48690f80-df85-11e9-9b30-16e9dd3c77e9.png)


@shashankbarsin - The `--image-name` argument is passed when we want to use an existing image. I don't think we should overload the imageName input with DockerHub case.

We are deriving the registry if it is not provided and dockerRegistryServiceConnection is present.

The user can pass registry input if they want to use a Docker login in some previous step (similar to Docker task) and for Docker Hub case.

Also, I think its better to have parity with --image-name, by keeping the input as imageName. 

Thoughts ?

Reply:
Take 2 inputs. `dockerRegistryServiceConnection` to point to the registry. And `image` input to include the namespace\repository within the container registry. If user wants to push to a public repository (let's say alpine), he specifies image value as 'alpine'. Let's say he wants to push to a private repository, then he can specify something like 'username/alpine'.

We can take the input the above mentioned way going ahead with all new container related tasks. About --image-name flag of func tool, I'd say having a consistent experience across all tasks is a higher priority than having consistency with func tool. Once the user specifies inputs in the above mentioned way, we can always compute the value to be passed on to --image-name

Reply:
As per offline discussion - 
- Removing imageName and registry inputs, and adding dockerHubNamespace input.We will currently not support deployments from an existing image.
- Making Docker service connection mandatory.